src/chapters/orm-intro/background.rst

```diff
@@ -0,0 +1,170 @@
+
+.. sourcecode:: javascript
+
+  "ConnectionStrings": {
+    "DefaultConnection": "server=localhost;userid=coding_events;password=PASSWORD!;database=coding_events;"
+  }
+
+The ``DefaultConnection`` property contains key/value pairs for each piece of information needed for our app to connect to the database created above. Be sure to replace ``PASSWORD`` with the password that you used when creating the ``coding_events`` user above.
+
+We now need to add a couple of NuGet packages to support our database connection. This process differs slightly for Windows and MacOS users. 
+
+Install MySQL Dependency
+########################
+
+Open the NuGet Package Manager in Visual Studio:
+
+- **Windows** - *Tools > NuGet Package Manager > Manage NuGet Packages for Solution*
+- **MacOS** - *Project > Manage NuGet Dependencies*
+
+Search for ``Pomelo.EntityFrameworkCore.MySql``. Select the package and install. This dependency provides code that is able to connect to a MySQL database from within an ASP.NET Core application using EF. Note that this package itself depends on the main EntityFrameworkCore package, ``Microsoft.EntityFrameworkCore.Relational``, so it is also installed.
+
+.. tip:: 
+
+   You can view installed packages and their dependencies by navigating to *Dependencies > NuGet* in the Solution Explorer and expanding a given package. 
+
+Install EF Core Tools
+#####################
+
+In this section, "terminal" refers to the Terminal app in MacOS and Powershell in Windows (use *Tools > Command Line > Developer Powershell* to open).
+
+Open a terminal and run:
+
+.. sourcecode:: bash
+
+   $ dotnet tool install -g dotnet-ef
+
+This command installs a set of command-line tools for working with EntityFrameworkCore *globally*, which means it will be available for any ASP.NET project we use in the future. We will use the tools provided by this package to update our database schema after adding or changing model classes. 
+
+.. note::
+
+   This not applies to *MacOS users only*.
+
+   For these tools to be accessible from the command line, they must be within your user path. Open ``~/.bash_profile`` and add the following line to the very bottom (recall that ``~`` is shorthand for your home directory, which is the directory you are in when you open a new terminal).
+
+   .. sourcecode:: bash
+
+      export PATH="$PATH:$HOME/.dotnet/tools/"
+
+   This will append the location of the EF tools to your user path. Save and close the file. Then close your terminal window and open a new one, so that the changes can take effect.
+
+To test that this install worked, run ``dotnet ef``. The output should be a message displaying basic EF tool commands and options.
+
+Once you have taken these steps, you are ready to set up the appropriate models and controllers for the application. We'll do that in the next section.
+
+.. index:: ! environment variables
+
+Key Takeaways
```

Comment:
a more targeted heading could be something like *Ensuring Connection Success and Security*